### ISSUE TITLE
Switch session store to one persistent across Heroku dynos

### ISSUE THREAD

**Issue by tsibley** [nextstrain/nextstrain.org#42]
Currently the file store is hitting Heroku's 24-hour max dyno lifetime, which is fairly annoying. Another session store should be configured so that sessions can persist for their entire lifetime despite Heroku's "helpful" cleansing. This adds a bit of complexity (another service/server) to something that is otherwise a dead-simple use of the filesystem, but so it goes.

**Comment by tsibley** [comment]
I looked at several low/no maintenance solutions for persistent, shared session storage including Memcached, Redis, and AWS DynamoDB.

Memcached and Redis have several choices of Heroku add-ons where we'll be able to operate within a free/dev tier. DynamoDB has a pricing model that made me weep, and that combined with it being a bit overkill for just sessions made me pass it over.

After getting Redis working, it turned out that the official offering I was using (like all the others) does not support encryption between Heroku and the service. The only option that turns out to do so is Memcachier's Memcached offering, so that's what I ended up choosing.

Note that is is fairly easy to swap these things out, so if Memcachier turns out to be no good for some unforeseen reason, we can swap it out. Choosing something cost-effective and low-maintenance is the more important part.

Steps:

- Provisioned the Heroku add-on [Memcachier](https://elements.heroku.com/addons/memcachier) which is a SaaS company running Memcached for folks. We're using the free/dev tier, which is limited to 10 connections and 25MB of memory. This should be plenty for our sessions for now, which are only ~500 bytes each.

- Added the `memcachier/memcachier-tls` buildpack to enable encrypted connections between our app on Heroku and Memcached by running `heroku buildpacks:add -i 1 memcachier/memcachier-tls`. This can also be done via the web admin interface. I'm not thrilled with the extra complexity this adds, including a poor failure condition if the background encrypted tunnel fails.

- Updated `authn.js` to optionally use Memcached for session storage if configured by the environment (6dff7d2).

- Removed session affinity add-on since it is no longer needed
